smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -673,7 +677,9 @@ private void generateOperationRequestSerializer(
             // Get the hostname, path, port, and scheme from client's resolved endpoint.
             // Then construct the request from them. The client's resolved endpoint can
             // be default one or supplied by users.
-            writer.write("const {hostname, protocol = $S, port, path: basePath} = await context.endpoint();", "https");
+
+            writer.addImport("requestBuilder", "rb", TypeScriptDependency.SMITHY_CORE);
```

Comment:
requestBuilder is from https://github.com/smithy-lang/smithy-typescript/pull/1107